Adding dspawpy

m3lab-zzl opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| java            | `@conda-forge/help-java`      |
| nodejs          | `@conda-forge/help-nodejs`    |
| c/c++           | `@conda-forge/help-c-cpp`     |
| perl            | `@conda-forge/help-perl`      |
| Julia           | `@conda-forge/help-julia`     |
| ruby            | `@conda-forge/help-ruby`      |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Gitter channel][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://gitter.im/conda-forge/conda-forge.github.io
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

m3lab-zzl commented:
@conda-forge/help-python 

Hi, the checks got stuck somehow, and I can not see its detialed error message, what happened?

bernt-matthias commented:
`ValueError: License file given in about/license_file (/home/conda/staged-recipes-copy/recipes/dspawpy/https://github.com/m3lab-zzl/staged-recipes/LICENSE.TXT) does not exist in source`

m3lab-zzl commented:
> `ValueError: License file given in about/license_file (/home/conda/staged-recipes-copy/recipes/dspawpy/https://github.com/m3lab-zzl/staged-recipes/LICENSE.TXT) does not exist in source`

Hi @bernt-matthias , thank you for the feedback

now only linux platform failed to pass check, could you update the error message?

bernt-matthias commented:
You may find it yourself by clicking 

- `Details` next to the failing test 
- Click the link saying `1 errors / 0 warnings`
- There you find some icons with a red x marking failed jobs .. click on them
- There you will find more red x(s) marking failing tests .. click on them  (`Run docker build` in your case)

bernt-matthias commented:
The problem is that the license file needs to be in the recipe dir. In the end it would be much better to have the LICENSE file packaged in the pypi package.

bernt-matthias reviewed (changes requested):
- Description fields are missing.
- Is there a source repository that you can refer to?

Since the docs are in Chinese I have no idea what the package is doing. 

Wondering about the license https://marketplace.huaweicloud.com/intl/product/OFFI873765584571371520

m3lab-zzl commented:
> * Description fields are missing.
> * Is there a source repository that you can refer to?
> 
> Since the docs are in Chinese I have no idea what the package is doing.
> 
> Wondering about the license https://marketplace.huaweicloud.com/intl/product/OFFI873765584571371520

Thanks for the detailed explanation, we will add a LICENSE file to pypi package soon. 

The `dspawpy` package is mainly for post-processing data generated by DS-PAW (a Density Functional Theory simulation package), we also will add an English description soon.

Again, thank you for your advice.

jaimergp commented on LICENSE.txt:
This is the license of the repository. Please do not modify .

You need to add the package license next to your `meta.yaml` file if not already distributed with the source.

m3lab-zzl replied:
Sorry for the mistake, I will change the original LICENSE back.

beenje commented:
Recipe looks good to me.

I noticed that version 1.0.8 is on pypi and includes a LICENSE.txt file.
You might want to update this PR by removing the local `LICENSE` file you created and referencing the file including in the sdist?

Otherwise, if you really want to have 1.0.4 on conda-forge, we can merge as it is and you can update the recipe for the new version after.

m3lab-zzl commented:
> Hi! This is the friendly automated conda-forge-linting service.
> 
> I was trying to look for recipes to lint for you, but couldn't find any. Please ping the 'conda-forge/core' team (using the @ notation in a comment) if you believe this is a bug.

Hi @beenje, I removed local LICENSE file and updated version info in the yaml file.

Please merge it if everything looks good to you.

beenje commented on recipes/dspawpy/meta.yaml:
```diff
@@ -0,0 +1,45 @@
+  home: "http://www.hzwtech.com/"
+  license: MIT
+  license_family: MIT
+  license_file: LICENSE
```
This should match the filename in the sdist:

```suggestion
  license_file: LICENSE.txt
```

m3lab-zzl commented:
Hi @beenje, I updated this `dspawpy` package to v1.1.0 on [Pypi](https://pypi.org/project/dspawpy/) several days ago. I thought conda may have automatically updated `dspawpy`, but it is hosting [old version of 1.0.8](https://anaconda.org/conda-forge/dspawpy).

How should I do to make conda follow the latest version published on Pypi? Thank you.

beenje commented:
> Hi @beenje, I updated this `dspawpy` package to v1.1.0 on [Pypi](https://pypi.org/project/dspawpy/) several days ago. I thought conda may have automatically updated `dspawpy`, but it is hosting [old version of 1.0.8](https://anaconda.org/conda-forge/dspawpy).
> 
> How should I do to make conda follow the latest version published on Pypi? Thank you.

The bot made some PRs to update the version that you, as maintainer of the feedstock, have to review and merge: https://github.com/conda-forge/dspawpy-feedstock/pulls

Please check https://conda-forge.org/docs/maintainer/updating_pkgs.html# for more information.

If you are confident that the recipe won't need changes on new release (like no changes in the dependencies), you can ask the bot to automerge those PRs. Check the information in of the bot PR on how to do that.

bernt-matthias commented:
Hi @beenje it does over here: https://github.com/conda-forge/dspawpy-feedstock/pulls

As a recipe maintainer @m3lab-zzl can take care of these.

m3lab-zzl commented:
Thank you.

I see there are two PRs open, but how do I merge them?

It seems that I do not have write permission. What do you mean that I can ask the bot to automerge them? Can you show me an example?

bernt-matthias commented:
> I see there are two PRs open, but how do I merge them?

There should be a merge button on the bottom of the PR. 

> What do you mean that I can ask the bot to automerge them? 

Note that this will also automerge future PRs. Never did this. But step 4 of the PR description seems to describe this quite well.

m3lab-zzl commented:
Hi @bernt-matthias, I am confused, there is no such button:
![image](https://github.com/conda-forge/staged-recipes/assets/102862066/05ea3180-52ea-43b8-b501-36dc0cfa1076)

m3lab-zzl commented:
The invitation mail was probably sent to an old email address that I have not logged in for a long time. I just checked all my email accounts, found nothing related (The policy of Outlook is to delete spam emails in 10 days by default).

Is it possible to re-send the mail to my default email account, I will watch for that this time.

bernt-matthias commented:
Not sure. I guess its best to ask at the conda-forge gitter channel.

m3lab-zzl commented:
Is there any work-around? Can I add a new contributor in a new version, such as v1.0.9 or v1.1.0?
